GH-43809: [Docs] Update extension type examples to not use UUID (#44120)
apache/arrow: docs/source/python/extending_types.rst

Defining extension types ("user-defined types")
-----------------------------------------------

Arrow affords a notion of extension types which allow users to annotate data
types with additional semantics. This allows developers both to
specify custom serialization and deserialization routines (for example,
to :ref:`Python scalars <custom-scalar-conversion>` and
:ref:`pandas <conversion-to-pandas>`) and to more easily interpret data.

In Arrow, :ref:`extension types <format_metadata_extension_types>`
are specified by annotating any of the built-in Arrow data types

> (the "storage type") with a custom type name and, optionally, a byte
> array that can be used to provide additional metadata (referred to as

Reviewer: Since it's `bytes` not `bytearray`, maybe we should do this:
```suggestion
(the "storage type") with a custom type name and, optionally, a
bytestring that can be used to provide additional metadata (referred to as
```

Author: Fair enough. A `bytearray` in python is mutable whereas a `bytes` is not. So seems fine.

            assert pa.types.is_integer(storage_type[0].type)
            assert storage_type[0].type == storage_type[1].type
            assert serialized == b""

            # return an instance of this subclass
            return RationalType(storage_type[0].type)

The special methods ``__arrow_ext_serialize__`` and ``__arrow_ext_deserialize__``
define the serialization and deserialization of an extension type instance.

This can now be used to create arrays and tables holding the extension type::

    >>> rational_type = RationalType(pa.int32())
    >>> rational_type.extension_name
    'my_package.rational'
    >>> rational_type.storage_type
    StructType(struct<numer: int32, denom: int32>)

    >>> storage_array = pa.array(
    ...     [
    ...         {"numer": 10, "denom": 17},
    ...         {"numer": 20, "denom": 13},
    ...     ],
    ...     type=rational_type.storage_type,
    ... )
    >>> arr = rational_type.wrap_array(storage_array)
    >>> # or equivalently
    >>> arr = pa.ExtensionArray.from_storage(rational_type, storage_array)
    >>> arr
    <pyarrow.lib.ExtensionArray object at 0x1067f5420>
    -- is_valid: all not null
    -- child 0 type: int32

>     [
>         10,
>         20
>     ]

Reviewer: ```suggestion
      [
        10,
        20
      ]
```

Author: This is actually a problem in several places in the documentation. It seems a lot of people's formatters assume 4 spaces when pyarrow always prints out 2.

    -- child 1 type: int32

>     [

Reviewer: ```suggestion
      [
```

        17,
        13

>     ]

Reviewer: ```suggestion
      ]
```
